hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
@racardoso—The output of that lowering is ill-formed vasm (we just happen not to run the `checkWidths()` pass after that point, I guess?).  By using `movb{}`, you've constrained `tmp1` and `tmp2` to be Vreg8's.  You then attempt to use them as Vreg64's, which is not allowed.